controller/src/beerocks/cli/beerocks_cli_bml.cpp

```diff
@@ -1806,5 +1913,63 @@ int cli_bml::get_dcs_scan_results(const std::string &radio_mac, unsigned int max
 {
     std::cout << "get_dcs_scan_results" << std::endl;
 
+    if (max_results_size == 0) {
+        std::cout << __func__ << "invalid input, max_results_size==0" << std::endl;
+        return -1;
+    }
+
+    struct BML_DCS_NEIGHBOR_AP *results = nullptr;
+
+    results =
+        (struct BML_DCS_NEIGHBOR_AP *)calloc(max_results_size, sizeof(struct BML_DCS_NEIGHBOR_AP));
+    if (results == nullptr) {
+        std::cout << __func__ << "failed allocate BML_DCS_NEIGHBOR_AP[" << max_results_size << "]"
+                  << std::endl;
+        return -1;
+    }
+
+    uint8_t status             = 0;
+    unsigned int results_count = max_results_size;
+    bool single_scan           = (is_single_scan == 1);
+    int ret = bml_get_dcs_scan_results(ctx, radio_mac.c_str(), &results, &results_count, &status,
+                                       single_scan);
+
+    if (ret == BML_RET_OK) {
+        if (results_count > max_results_size) {
+            std::cout << __func__ << "ERROR: results_count(" << results_count << ")"
+                      << " > max_results_size(" << max_results_size << ")" << std::endl;
+        } else if (status == 0 && results_count > 0) {
+            //todo: replace the temporary debug print with actual prints of the new results structure
```

Comment:
I'm not sure I follow - what is temporary with the debug print? Aren't the results already in the new structure?

Reply:
no - since the PR that changes the structure is not available yet
(PRs 4/5): 
https://github.com/prplfoundation/prplMesh/pull/694